### PR TITLE
Fix false negative in IgnoredReturnValue

### LINKED ISSUE #3170: False negative in IgnoredReturnValue
## Expected Behavior
A use of an annotated method with `@CheckReturnValue` at the end of an if expression should be flagged by `IgnoredReturnValue` if the output of the if is not used.

Same for the `when`.
## Steps to Reproduce
```kotlin
package test

import kotlin.random.Random

@CheckReturnValue
fun returnsInt() = 42

if (Random.nextBoolean()) {
    println("hello")
} else {
    returnsInt()
}
```

## Your Environment
* Version of detekt used: master (1.14.2)

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #3170

### CODE REVIEW

File: detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt [hunk 1]
```diff
@@ -72,8 +65,12 @@ class IgnoredReturnValue(config: Config = Config.empty) : Rule(config) {
     @Suppress("ReturnCount")
     override fun visitCallExpression(expression: KtCallExpression) {
         super.visitCallExpression(expression)
-
         if (bindingContext == BindingContext.EMPTY) return
+
+        val parentQualifiedExpression = expression.getStrictParentOfType<KtQualifiedExpression>()
+        if (parentQualifiedExpression != null && parentQualifiedExpression.selectorExpression != expression) return
+        if (expression.isUsedAsExpression(bindingContext)) return
```
Comment (BraisGabin) [inline detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt:72]: 🤔 what if the statment is not the last of the if? I mean:

```kotlin
val a = if (random()) {
  2
} else {
  returnsInt()
  3
}
```

Is this handled correctly?

Comment (t-kameyama) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt:72]: Yes 5c2d3e3, but the following case is not flagged. I will fix.
```kotlin
if (Random.nextBoolean()) {
    1
} else {
    returnsInt()
    2
}.plus(1)
```

Comment (t-kameyama) [reply detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValue.kt:70]: Fixed in 217901b

**Review by BraisGabin — APPROVED** [review]
👏🏻👏🏻

**Review by schalkms — APPROVED** [review]

**Pull request merged**